src/java/org/jivesoftware/openfire/net/SASLAuthentication.java

```diff
@@ -192,6 +192,24 @@ public static Element getSASLMechanismsElement( ClientSession session )
     {
         final Element result = DocumentHelper.createElement( new QName( "mechanisms", new Namespace( "", SASL_NAMESPACE ) ) );
         for (String mech : getSupportedMechanisms()) {
+            if (mech.equals("EXTERNAL")) {
+                boolean trustedCert = false;
+                if (session.isSecure()) {
+                    final LocalClientSession localClientSession = (LocalClientSession)session;
+                    if (localClientSession != null) {
+                        final Connection connection = localClientSession.getConnection();
+                        final KeyStore keyStore = connection.getConfiguration().getIdentityStore().getStore();
+                        final KeyStore trustStore = connection.getConfiguration().getTrustStore().getStore();
+                        final X509Certificate trusted = CertificateManager.getEndEntityCertificate(connection.getPeerCertificates(), keyStore, trustStore);
+                        if (trusted != null) {
+                            trustedCert = true;
```

Comment:
Because continue just there isn't enough, and I decided I'd rather have a single "continue". Nothing more than a stylistic choice really.